torchtune/data/_utils.py

```diff
@@ -142,3 +150,131 @@ def format_content_with_images(
             final_content_list.append({"type": "image", "content": images.pop(0)})
 
     return final_content_list
+
+
+def chain(*funcs: Callable):
+    """
+    Chain a list of functions together into a single function.
+
+    Args:
+        funcs (List[Callable]): list of functions to chain together
+
+    Returns:
+        Callable: chained function
+    """
+
+    def chained_fn(x):
+        for fn in funcs:
+            x = fn(x)
+        return x
+
+    return chained_fn
+
+
+@requires_torchdata
+def load_hf_dataset(
+    source: str,
+    transform: Transform,
+    filter_fn: Optional[Callable] = None,
+    shuffle: bool = True,
+    seed: int = 0,
+    num_workers: int = 1,
+    parallel_method: Literal["process", "thread"] = "thread",
+    **load_dataset_kwargs: Dict[str, Any],
+) -> DatasetType:
+    from torchdata.nodes import IterableWrapper, ParallelMapper, SamplerWrapper
+
+    # Need to lazy import to avoid circular dependency
+    from torchtune.training._distributed import get_world_size_and_rank
```

Comment:
A minor point but we should avoid doing this

Reply:
Lazy importing? I think the workaround would be to up-level torchtune.training._distributed to eg torchtune.distributed. WDYT?

Reply:
Fyi our hero @joecummings addressed this in #2155. You can now do `from torchtune.utils import get_world_size_and_rank` at the top of the file